### ISSUE TITLE
python run_eval.py     error

### ISSUE THREAD

**Issue by kk6398** [prs-eth/Diffusion-Super-Resolution#7]
Hi, thanks for your excellent job. However, when I run python run_eval.py according to your statement, I have encountered the following error：
![1689065793120](https://github.com/prs-eth/Diffusion-Super-Resolution/assets/70420660/4d77009b-8a52-4217-95a6-8360102a87af)
![1689065840692](https://github.com/prs-eth/Diffusion-Super-Resolution/assets/70420660/a2a2c1d2-a376-418d-8129-dfc718aaa53a)

**Comment by nandometzger** [comment]
Hi, thanks for your interest!

I quickly tried to reproduce your issue. I did indeed spot a small bug, which I corrected by adding this else statement: https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/model/gad_base.py#L52
After this fix, the script is working fine for me: 
![image](https://github.com/prs-eth/Diffusion-Super-Resolution/assets/42088121/753ca877-3c2f-4bf1-b963-97cb03c5ee82)

However, it is unlikely that this is the true source of your error. since your report suggests that the variable `output` is a `None`. Which seems weird to me. Did you check if the dataloaders are outputting valid samples and if the data locations are set up properly? 

Please let me if the problem still persists! Then I would like to fix it.
Best
Nando

**Comment by kk6398** [comment]
> 
Thank you for your prompt reply！
I try to output the dataloaders、the model and the sample, the output content in order is：
dataloaders: self.dataloader===== <torch.utils.data.dataloader.DataLoader object at 0x7fd64f6da190>
model:
![image](https://github.com/prs-eth/Diffusion-Super-Resolution/assets/70420660/bbd3eb47-fa50-4c9e-8a54-67f5396e8d96)
sample:  
![image](https://github.com/prs-eth/Diffusion-Super-Resolution/assets/70420660/5e429744-0d5d-484e-9f32-10a2e9cae4cf)

And I try to output the "output",   Indeed, as you said, it is ”None“

**Comment by nandometzger** [comment]
hmm... inputs seem to be fine...
... as mentioned before, I cannot reproduce the error although I am using the same code as in this repository. Other issues in this repository suggested they can also run it. Are you sure you didn't do any custom modifications? for example, I see that your `get_loss()` statement is on line 43, while in the original repository, it was on line 42 (https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/run_eval.py#L42). Which modification did you do exactly? Can you revert them maybe and see if it works?

**Comment by kk6398** [comment]
> hmm... inputs seem to be fine... ... as mentioned before, I cannot reproduce the error although I am using the same code as in this repository. Other issues in this repository suggested they can also run it. Are you sure you didn't do any custom modifications? for example, I see that your `get_loss()` statement is on line 43, while in the original repository, it was on line 42 (
> 
> https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/run_eval.py#L42
> 
> ). Which modification did you do exactly? Can you revert them maybe and see if it works?

Thank you for your reply！I tried to recover all the code, but the same error occurred. 
Therefore, I have to verify the loading of the dataset again: I found that when running "Python scripts/create_diml_npy. py./datafolder/DIML", only/DIML/npy/* HR and no LR related files will appear. Is this the reason for the error?
![image](https://github.com/prs-eth/Diffusion-Super-Resolution/assets/70420660/df755b4f-9c42-46c6-821c-435359ac28b1)

**Comment by nandometzger** [comment]
I checked my folder structure and data. It is the exact same as yours.

Did you try looking at the `y_pred` variable right after the diffusion. about here: https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/model/gad_base.py#L57

Also, which python version do you use? It would be 3.9 like in the requirements. Some older versions do not support the dict merging operator here:
https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/model/gad_base.py#L62
try using `return  {**{'y_pred': y_pred}, **aux}` instead maybe?

**Comment by kk6398** [comment]
> I checked my folder structure and data. It is the exact same as yours.
> 
> Did you try looking at the `y_pred` variable right after the diffusion. about here:
> 
> https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/model/gad_base.py#L57
> 
> Also, which python version do you use? It would be 3.9 like in the requirements. Some older versions do not support the dict merging operator here:
> 
> https://github.com/prs-eth/Diffusion-Super-Resolution/blob/881a1dc0049e17cf9d49e6def98dc6a6024a01a3/model/gad_base.py#L62
> 
> 
> try using `return {**{'y_pred': y_pred}, **aux}` instead maybe?

Really thank you very much！”return  {**{'y_pred': y_pred}, **aux}“ deal this issue! You are my savior! 
And I verify the results in paper, which is correct.

Emm, my research direction is 3D reconstruction,so I am a bit unfamiliar with GDSR. 
I want to use the GDSR method to improve the quality of depth maps and thereby enhance the performance of 3D reconstruction. Therefore, how can I obtain visualized results of this repo？
Additionally, I would like to add a question, can I obtain a x2 model through training?

**Comment by nandometzger** [comment]
hi,
DADA for 3D reconstruction sounds exciting =)
yes, obtaining an x2 model is possible through training. It is as easy as taking the training code and using the flag `--scaling 2` for x2 upsampling. Make sure to adjust the learning rate scheduler according to your dataset.
Note we did not test our model for x2. I assume that it will work very well. However, I cannot tell you with certainty that it will provide SOTA results.
